monitor.py

```diff
@@ -211,6 +215,14 @@ def run_once(self, progress):
 
         :param progress: A TimestampData representing the
            work done by the Monitor up to this point.
+
+        :return: A TimestampData representing how you want the
+            Monitor's entry in the `timestamps` table to look like
+            from this point on.  NOTE: Modifying the incoming
+            `progress` and returning it is generally a bad idea,
+            because the incoming `progress` is full of old
+            data. Instead, return a new TimestampData containing data
+            for only the fields you want to set.
```

Comment:
This turned out to be an important distinction in my circulation branch. Inside `TimelineMonitor.catch_up_from`s it's easier to modify the `progress` object you were given and return it, because `TimelineMonitor.run_once` will always set the timestamps appropriately. But for `Monitor` in general that won't work; you need to create a new TimestampData if you want to record achievements.

Reply:
That's what I do in `OPDSImportMonitor` below.